examples/Person/ffi/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+use std::ffi::CString;
+use std::ffi::CStr;
+
+pub struct Person {
+    name: String,
+    lucky_number: i32,
+}
+
+impl Person {
+    fn new(name: &str, lucky_number: i32) -> Person {
+        Person {
+            name: String::from(name),
+            lucky_number: lucky_number,
+        }
+    }
+
+    fn get_name(&self) -> String {
+        String::from(&self.name)
+    }
+
+    fn get_lucky_number(&self) -> i32 {
+        self.lucky_number
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn person_new(_class: *const i8, name: *const i8, lucky_number: i32) -> *mut Person {
+    let name = unsafe {
+        CStr::from_ptr(name).to_string_lossy().into_owned()
+    };
+    Box::into_raw(Box::new(Person::new(&name, lucky_number)))
+}
+
+#[no_mangle]
+pub extern "C" fn person_name(p: *mut Person) -> *const i8 {
+    let p = unsafe { &*p };
+    let name = CString::new(p.get_name()).unwrap();
+    let ptr = name.as_ptr();
+    std::mem::forget(name);
```

Comment:
This PR fixes #5, but it introduces a new memory leak with this `std::mem::forget` call lol.